plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcConnectorTest.java

```diff
@@ -132,6 +134,15 @@ protected Optional<DataMappingTestSetup> filterDataMappingSmokeTestData(DataMapp
         return Optional.of(dataMappingTestSetup);
     }
 
+    @Override
+    protected Optional<SetColumnTypeSetup> filterSetColumnTypesDataProvider(SetColumnTypeSetup setup)
+    {
+        if (setup.sourceValueLiteral().equalsIgnoreCase("'char-to-varchar'")) {
+            return Optional.of(new SetColumnTypeSetup(setup.sourceColumnType(), setup.sourceValueLiteral(), setup.newColumnType(), "'" + padEnd("char-to-varchar", 100, ' ') + "'")); // H2 retains trailing spaces when converting char(n) columns to varchar(n)
```

Comment:
per https://github.com/trinodb/trino/pull/15641#discussion_r1071271274 -- remove. Fix the super class as a prep commnit.

Reply:
Then all of the other databases will fail.

i.e. PostgreSQL converts char(n) to varchar(n-m) by dropping m trailing spaces even if the source literal contains trailing spaces.

Reply:
Maybe it's related to the https://github.com/h2database/h2database/issues/3385 ?

Reply:
> PostgreSQL converts char(n) to varchar(n-m) by dropping m trailing spaces even if the source literal contains trailing spaces.

That is acceptable (or maybe even specwise).

Is your code comment about H2 indicating H2 is different in this regard? And that it retains all trailing spaces when converting from `char(n)` to `varchar(n-m)` ?

Reply:
Yes, exactly

Reply:
then how can it retain trailing whitespace when converting to a shorter varchar?
Can H2 return value of length m for varchar(n) column where m > n?

Reply:
no, it retains spaces up to the new length (trimming redundant spaces)

Reply:
so it sounds it behaves correctly . per https://github.com/trinodb/trino/pull/15641#discussion_r1071400490 we should fix the super class instead of overriding it here.

https://github.com/trinodb/trino/pull/15782